### PR TITLE
Fix hang on exit. 

### PR DESCRIPTION
**Pull request by jimfcarroll**
The VideoPlayer has a JobQueue that's not managed directly by the JobManager and so when the JobManager shuts down this queue can't be clearer. 

## Description

This change just makes the CJobQueue::IsProcessing method also check the CJobManager::m_isrunning flag since if it's not running then the JobQueue is also not processing.

## Motivation and Context

If you try to exit while playing a video, Kodi hangs.

## How Has This Been Tested?

Easily reproduced the problem. Made sure there was a clean exit after the change.

## Types of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves existing functionality)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the [Code guidelines](https://codedocs.xyz/xbmc/xbmc/code_guidelines.html) of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the [CONTRIBUTING](https://github.com/xbmc/xbmc/blob/master/CONTRIBUTING.md) document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: xbmc/utils/JobManager.h [hunk 1]
```diff
@@ -322,6 +322,7 @@ class CJobManager
 protected:
   friend class CJobWorker;
   friend class CJob;
+  friend class CJobQueue;
```
Comment (ksooo) [inline xbmc/utils/JobManager.h]: Really? Why do you not introduce a public method `CJobMananger::IsRunning`?

Comment (jimfcarroll) [reply xbmc/utils/JobManager.h]: I almost did that but I didn't for a few reasons: 

1) The m_isrunning state isn't used in other places you might expect like JobManager::IsProcessing so it seems to be really in internal implementation detail. 
2) The JobQueue and the JobManager are already completely dependent on each others implementation details so knowing that the `JobManager::m_isrunning` flag, when set to `false` means the `JobQueue::IsProcessing` should be false didn't seem like it needed a change to the public interface.

If people would prefer it the other way, I have no problem changing it.

Comment (ksooo) [reply xbmc/utils/JobManager.h]: Yeah, it's messy. I just thought "why make it even worse?"

Comment (jimfcarroll) [reply xbmc/utils/JobManager.h]: I think the point was the keep the mess internal rather than let it leak into the public interface that would need to be maintained going forward.

Comment (ksooo) [reply xbmc/utils/JobManager.h]: No hard feelings here. As it is working as it is, it's your decision. :-)

File: xbmc/utils/JobManager.cpp [hunk 2]
```diff
@@ -164,7 +164,7 @@ void CJobQueue::CancelJobs()
 
 bool CJobQueue::IsProcessing() const
 {
-  return !m_processing.empty() || !m_jobQueue.empty();
+  return CJobManager::GetInstance().m_running && (!m_processing.empty() || !m_jobQueue.empty());
```
Comment (FernetMenta) [inline xbmc/utils/JobManager.cpp]: @jimfcarroll  Those jobs must be executed or the app wonˋt get the corrct state. Discarding jobs is not a corect solution.

Comment (jimfcarroll) [reply xbmc/utils/JobManager.cpp]: Yeah. I thought about that too since that's what happens with the other JobQueues that are part of the JobManager. I did it this way though because 1) doing it by draining the queue seemed more involved and potentially error prone. 2) The JobManager gets turned to !m_isrunning only on shutdown. 3) If (2) were to ever change so that the state could go back to m_isrunning then the JobQueue would continue on.

TL;DR Ultimately it came down to the fact that this was trivial and foolproof.

I can work up a different solution if you'd prefer though. Just let me know.

Comment (FernetMenta) [reply xbmc/utils/JobManager.cpp]: Yes, please do. If we discard jobs, we lose i.e. resume point on exit. Maybe other more severe things too.
There must also be some other issue with JobQueue: start video player and halt execution by setting a breakpoint somewhere in player, do some investigation for a minite, then continue and stop playback -> stop hangs because jobqueue gets never empty.

Comment (jimfcarroll) [reply xbmc/utils/JobManager.cpp]: I did that. That's how I found the problem. It wasn't a guess:

The JobQueue in the VideoPlayer isn't managed in the JobManager so when the JobManager is stopped, that particular JobQueue still has Jobs in it. THEN, on shutdown, the VideoPlayer waits for its JobQueue to cease from processing (that is, it waits until the `CJobQueue::IsProcessing()` call to return `false`). But, it never will because the JobManager is stopped but entries remain.

So this change simply makes `CJobQueue::IsProcessing` *correctly* report that once the singleton `JobManager` is no longer running, it's no longer processing.

**Comment by iz8mbw** [comment]
Hi.
I don't know if it's related, but on Android Kodi crashes just after closing it, also after this PR (tested on Nvidia Shield TV).
After closing Kodi, Android inform that Application closed unexpectedly...

**Comment by jimfcarroll** [comment]
Hi @iz8mbw ,

@da-anda was looking into that. I don't know if it's related but I have more work to do on the Job stuff anyway so let me know here if you find anything interesting.

**Comment by da-anda** [comment]
on my latest test with this PR in, the crash on exit after I had played a video didn't happen for me anymore. But I only did one test and I could have been lucky. @koying mentioned that the logcat of the crash seems to be related to Android specific stuff, so I guess he's looking into it.

**Comment by iz8mbw** [comment]
Hi @koying @da-anda @jimfcarroll 
I tested about all the recents Kodi 18 nigthly and this issue of Kodi crash after closing (Android inform "Kodi Application closed unexpectedly") seems to be born since that version: kodi-20180113-9d123033-master-arm64-v8a (so after "Jan 11, 2018" or "Jan 12, 2018" GitHub PR/commits). That GitHub PR/commits was made to solve the "Kodi frozen when loading" issue.
I hope that can help to address Kodi crash after closing.
Thanks!
